waiter/src/waiter/statsd.clj

```diff
@@ -36,21 +36,61 @@
   [host-address]
   (InetAddress/getByName host-address))
 
-(let [config (atom nil)
-      socket-agent (agent nil)
-      environment (atom nil)
-      cluster (atom nil)
-      server (atom nil)
-      publish-interval-ms (atom 0)
-      histogram-max-size (atom 10000)]
+(defn keep-all-metrics
+  "default predicate for always sending all metrics"
+  [metric-group metric]
+  true)
+
+(defn- sanitize-host [host]
+  (resolve-ip host))
+
+(defn- sanitize-port [port]
+  (if (integer? port) port (Integer/parseInt port)))
+
+(defn- sanitize-predicate [predicate-fn-symbol]
+  (if predicate-fn-symbol
+    (utils/resolve-symbol! predicate-fn-symbol)
+    keep-all-metrics))
+
+(defn make-dd-client
+  "build non-blocking dogstatsd client"
+  [{:keys [host port]} {:keys [prefix]}]
+  (NonBlockingStatsDClient.
+      ^String (or prefix "waiter")
+      ^String host
+      ^Integer (sanitize-port port)))
+
+(defn make-dd-tag-config
+  "build string arrays for common datadog tags"
+  [{:keys [cluster environment server]}]
+  (let [cluster-tags (list (str "env:" environment)
+                           (str "cluster:" cluster))
+        server-tags (conj cluster-tags (str "server:" server))]
+    {:cluster-tags cluster-tags
+     :server-tags server-tags}))
```

Comment:
How do we distinguish between cluster and server tags?

Reply:
These are referenced in the :dogstatsd-client-fn. I added a comment on one of them.